fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java

```diff
@@ -1196,6 +1196,75 @@ public Partition dropPartition(long dbId, String partitionName, boolean isForceD
         return dropPartition(dbId, partitionName, isForceDrop, !isForceDrop);
     }
 
+    public Partition dropPartitionForTruncate(long dbId, boolean isForceDrop,
+                                            RecyclePartitionParam recyclePartitionParam) {
+        // 1. If "isForceDrop" is false, the partition will be added to the Catalog Recyle bin, and all tablets of this
+        //    partition will not be deleted.
+        // 2. If "ifForceDrop" is true, the partition will be dropped immediately
+        Partition partition = recyclePartitionParam.partition;
+        if (partition != null) {
+            idToPartition.remove(partition.getId());
```

Comment:
need add `nameToPartition.remove(partitionName)`,  just like  OlapTable.dropPartition function.

so refactor dropPartition and dropPartitionForTruncate to reuse code;

Reply:
replacePartition had already removed old partition from nameToPartition and added new partition created by truncate process. so it should not be called in truncate process.

Reply:
ok,  this had resolved